### DISCUSSION TITLE
Mortality

### DISCUSSION THREAD

**Discussion by krosenfeld-IDM** [InstituteforDiseaseModeling/laser#25] [General]
From [Open Design Questions](https://github.com/InstituteforDiseaseModeling/laser/wiki/Open-Design-Questions)
>  Jonathan has been modeling mortality by pre-fitting curves for the EULA populations by node and using pre-calculated lifespans for the modeled population.

original author @jonathanhhb

**Comment by krosenfeld-IDM** [comment]
@KevinMcCarthyAtIDM response:
> Lots of options here, and maintaining flexibility is important. I think an approach where we have a "mortality_step" in the list of model steps, that the user fills in, is a good general reference design. For models of a certain scale, filling a priority queue seems a good option to have few "steps per death" (1 random draw, one pop off of the queue). Giving agents a "date of death" property seems like another one, 1 random draw though potentially many conditional checks. But I don't want to have to foresee things - if people want to have an agent property like "malnutrition" that raises mortality risk in a way we won't know before running the sim, then they should be able to have a more flexible approach. Or say I want a birth rate = mortality rate constant population, I might want to implement "mortality" as just a reset to a newborn state.

**Comment by jonathanhhb** [comment]
Sounds like there is nothing blocking here (design-wise) and that we'll probably want to allow, support and even have examples for 2 or 3 ways of doing mortality.
